### PR TITLE
feat: [MTL] Initial commit to open source

### PR DESCRIPTION
**Pull request by koktong-ong**
SBL MTL initial commit to open source includes
Platform and Silicon package

### CODE REVIEW

File: Silicon/MeteorlakePkg/Library/IgdOpRegionLib/IgdOpRegionLib.inf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#
+#  Copyright (c)  1999  - 2014, Intel Corporation. All rights reserved
+#
+# This program and the accompanying materials are licensed and made available under
+# the terms and conditions of the BSD License that accompanies this distribution.
+# The full text of the license may be found at
+# http://opensource.org/licenses/bsd-license.php.
+#
+# THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+# WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
```
Comment (gdong1) [inline Silicon/MeteorlakePkg/Library/IgdOpRegionLib/IgdOpRegionLib.inf:11]: The license need change to BSD plus one. Pleas check if you could use common lib Silicon\CommonSocPkg\Library\IgdOpRegionLib instead.

File: Silicon/MeteorlakePkg/Library/BootGuardLibCBnT/BootGuardLibCBnT.inf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  Component description file for Boot Guard Library.
+
+  Copyright (c) 2023, Intel Corporation. All rights reserved.<BR>
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
```
Comment (gdong1) [inline Silicon/MeteorlakePkg/Library/BootGuardLibCBnT/BootGuardLibCBnT.inf:11]: Need use BSD plus license. Please check if you could use common lib Silicon\CommonSocPkg\Library\BootGuardLibCBnT

File: Silicon/MeteorlakePkg/Include/Library/IgdOpRegionLib.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  Acronyms:
+    IGD:        Internal Graphics Device
+    NVS:        ACPI Non Volatile Storage
+    OpRegion:   ACPI Operational Region
+    VBT:        Video BIOS Table (OEM customizable data)
+    IPU:        Image Processing Unit
+
+  Copyright (c) 2021, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
```
Comment (gdong1) [inline Silicon/MeteorlakePkg/Include/Library/IgdOpRegionLib.h:15]: Please check if you could use common header file Silicon\CommonSocPkg\Include\Library\IgdOpRegionLib.h

File: Silicon/MeteorlakePkg/Include/Library/VmdInfoLib.h [hunk 4]
```diff
@@ -0,0 +1,141 @@
+/** @file
+  @ Header file to get VMD information.
+
+  Copyright (c) 2023, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Include/Library/VmdInfoLib.h:5]: there are some defines in VmdInfoLib.h being used in ACPI files (eg. MAX_OS_VISIBLE_BUSES_WITH_VMD refer by HostBus.asl)

File: Silicon/MeteorlakePkg/Include/Library/BootGuardLib.h [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+
+@par Specification Reference:
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Include/Library/BootGuardLib.h]: patch submitted with removed duplicate lib

File: Silicon/MeteorlakePkg/Library/PchSpiLib/PchSpiLib.inf [hunk 6]
```diff
@@ -0,0 +1,35 @@
+## @file
+#
+#  Copyright (c) 2020 - 2022, Intel Corporation. All rights reserved.<BR>
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+##
+
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Library/PchSpiLib/PchSpiLib.inf]: patch submitted with removed duplicate lib

File: Silicon/MeteorlakePkg/Library/MeChipsetLib/MeChipsetLib.inf [hunk 7]
```diff
@@ -0,0 +1,41 @@
+## @file
+#
+#  Copyright (c) 2020 - 2022, Intel Corporation. All rights reserved.<BR>
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Library/MeChipsetLib/MeChipsetLib.inf]: patch submitted with removed duplicate lib

File: Silicon/MeteorlakePkg/Library/IgdOpRegionLib/IgdOpRegionLib.inf (hunk 1 above)
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Library/IgdOpRegionLib/IgdOpRegionLib.inf]: patch submitted with removed duplicate lib

File: Silicon/MeteorlakePkg/Library/BootGuardLibCBnT/BootGuardLibCBnT.inf (hunk 2 above)
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Library/BootGuardLibCBnT/BootGuardLibCBnT.inf]: patch submitted with removed duplicate lib

File: Silicon/MeteorlakePkg/Include/Library/IgdOpRegionLib.h (hunk 3 above)
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Include/Library/IgdOpRegionLib.h]: patch submitted with removed duplicate lib

File: Silicon/MeteorlakePkg/Include/Library/VmdInfoLib.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Include/Library/VmdInfoLib.h]: removed VmdInfoLib.h and moved required the defines to asl file

File: Silicon/MeteorlakePkg/Include/Register/CmosMap.h [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  CMOS 0x80 - 0xFF are reserved for OEM use only
+*/
+
+
+//
+// PLATFORM SPECIFIC USAGE
+//
+#define FIT_REC_TXT_POLICY_TYPE_A 0x2A
+
+#define FIT_REC_TXT_POLICY_TYPE_A 0x2A
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Include/Register/CmosMap.h]: patch submitted

File: Silicon/MeteorlakePkg/Include/Register/CmosMap.h [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+#define RTC_ADDRESS_CENTURY       0x32
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Include/Register/CmosMap.h]: patch submitted

File: Silicon/MeteorlakePkg/Library/PsdLib/PsdLib.c [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+
+  Copyright (c) 2020-2023, Intel Corporation. All rights reserved.<BR>
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Library/PsdLib/PsdLib.c]: Done update the file license

File: Silicon/MeteorlakePkg/Library/PsdLib/PsdLib.inf [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+#
+#  Copyright (c) 2020-2023, Intel Corporation. All rights reserved.<BR>
+#  This program and the accompanying materials
+#  are licensed and made available under the terms and conditions of the BSD License
+#  which accompanies this distribution.  The full text of the license may be found at
+#  http://opensource.org/licenses/bsd-license.php
+#
+#  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+#
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Silicon/MeteorlakePkg/Library/PsdLib/PsdLib.inf:5]: Done update the file license

**Review by gdong1 — APPROVED** [review]

**Pull request merged**